src/main.rs

```diff
@@ -647,116 +632,146 @@ fn main() {
                     run_pin.set_high();
                 }
                 if frame_acquire {
-                    info!("Connecting to frame socket {}", address);
-
-                    match SocketStream::from_address(&address, *&config.use_wifi) {
-                        Ok(mut stream) => {
-                            info!("Connected to {}, waiting for frames from rp2040", if config.use_wifi { &"tc2-frames server"} else { &"thermal-recorder unix socket"});
-                            let mut sent_header = false;
-                            let mut ms_elapsed = 0;
-                            'send_loop: loop {
-                                // Check if we need to reset rp2040 because of a config change
-                                if let Ok((_,is_audio)) = restart_rx.try_recv() {
-                                    frame_acquire = !is_audio;
-
-                                    cross_thread_signal_2.store(true, Ordering::Relaxed);
-                                    loop {
-                                        info!("Restarting rp2040");
-                                        if !run_pin.is_set_high() {
-                                            run_pin.set_high();
-                                            sleep(Duration::from_millis(1000));
-                                        }
+                    info!("Connecting to frame sockes");
+                    let mut ms_elapsed = 0;
+                    loop {
+                        let mut connections = 0;
```

Comment:
I'd probably write this more idiomatically as something like:
```
for (index, (address, mut stream)) in sockets.iter_mut().enumerate().filter(|(_, (_, stream))| stream.is_none()) {
    let is_frame_server_socket = index == 0;
    let is_wifi_debug_socket = if is_frame_server_socket { *&config.use_wifi } else { false };
    let stream_connection = SocketStream::from_address(&address, is_wifi_debug_socket).ok();
    if stream_connection.is_some() {
        println!("Connected to {}", &address);
    }
    stream = stream_connection
}
let connections = sockets.iter().filter(|(_, stream)| stream.is_some()).count();
```

Also, the thing about seeing if it's the first socket in the array and applying special logic to it in that case is a bit weird – would be better to store that info in the tuple being iterated IMO.